### ISSUE TITLE
Update SC's social sharing information

### ISSUE THREAD

**Issue by Aminah-SC** [suttacentral/suttacentral#1486]
Following on from:

https://github.com/suttacentral/suttacentral/issues/1400

We now need to assign the generic og (open graphic) meta tag properties used by social sharing platforms and, if necessary, platform-specific tags as well. See the following for more on og here:

http://ogp.me/
https://www.aira.net/open-graph-social-media-traffic/

#### Acceptance criteria (the list of things that need to be done for the ticket to be considered finished):

- [ ] social sharing image has been updated
- [ ] relevant meta tags have been added to sutta `<head>` eg:

    ```
    <meta property="og:title" content="SUTTA TITLE">   
    <meta property="og:description" content="SUTTAPLEX BLURB">
    <meta property="og:image" content="PATH/SOCIAL SHARING IMAGE.png>
    ```

Has been tested on:

- [ ] Discourse
- [ ] Pinterest
- [ ] FB
- [ ] LinkedIn
- [ ] Twitter
- [ ] Whatsapp

**Comment by thesunshade** [comment]
The fix should also be tested for Discourse's "inline Onebox". When you include a url in a paragraph of text on the forum, Discourse should replace the url with the linked title of the page.

You probably don't need it, but here is the code on Discourse that does the magic: https://github.com/discourse/discourse/blob/main/lib/inline_oneboxer.rb

Again, you probably don't need it, but here is the Discourse thread where this feature was developed (from 2017). There they are calling it "miniOnebox"

**Comment by gnlaera** [comment]
How to upvote???

**Comment by ihongda** [comment]
As pointed out here:
https://stackoverflow.com/a/34284383/15141877
`Facebook does not parse JavaScript at all, you can´t use dynamic Open Graph tags. It does not really make sense to change them on the fly anyway.`

（Looks like linkin and twitter are the same）

So even if we update the tag information related to open graph, Facebook will still only grab the default open graph content.

**Comment by ihongda** [comment]
![深度截图_选择区域_20210929172147](https://user-images.githubusercontent.com/4037966/135241100-580510e9-189c-445e-8b79-be4abcb4d0a7.png)

This may be a problem with the localization file loading. It is normal in my local area. Let me check

The others i am modifying

**Comment by sujato** [comment]
I am checking the OG metadata on the pages, it looks good.

But it's not showing up, I guess due to the social media sites not executing js. So maybe we should do SSR? Send the `<head>` as prerendered HTML?

https://github.com/lit/lit/tree/main/packages/labs/ssr

**Comment by agilgur5** [comment]
> But it's not showing up, I guess due to the social media sites not executing js. So maybe we should do SSR? Send the `<head>` as prerendered HTML?

Yes, that's the conventional way of doing it for crawlers that don't support JS (or users that don't enable JS - see also https://github.com/benmneb/suttacentral-static).

It splits into a few approaches:
- SSG (static site generation) - Pre-generate the whole site or parts of the site (https://suttacentral.express is one variation of this)
- SSR (server side rendering) - Have the back-end server run some JS to generate more parts of the HTML per page, partly overlapping with the client-side JS router
  - or, in this case, partly rewrite some bits of the rendering in Python

There is some overlap between the two terms (SSG could be viewed as a pre-generated subset of SSR. these days there's also Server Actions too). Both can also support partial rendering, where the client-side JS can "rehydrate" the HTML to make it more interactive or add more HTML that requires JS. Both can also be cached to varying degrees.

I'd suggest more of the former, as it would be less load on the server and wouldn't require the current Python back-end to run any JS. It would, however, require more automation to pre-generate HTML previews for various URLs.
That being said, if only purely for populating meta tags, a small amount of logic could be re-implemented in the Python to just add those. That may or may not be worthwhile as a stop-gap measure

<hr />

Also, for reference, back-linking [some](https://discourse.suttacentral.net/t/better-preview-cards-for-suttas-in-external-environments-such-as-discourse-and-discord/41646?u=agilgur5) [forum](https://discourse.suttacentral.net/t/link-previews-still-broken-upvote-on-github/21252?u=agilgur5) [posts](https://discourse.suttacentral.net/t/generic-suttacentral-oneboxes/12048?u=agilgur5) about this issue
